sql/core/src/main/scala/org/apache/spark/sql/internal/columnNodeSupport.scala

```diff
@@ -250,7 +252,8 @@ private[sql] object ColumnNodeToExpressionConverter extends ColumnNodeToExpressi
 /**
  * [[ColumnNode]] wrapper for an [[Expression]].
  */
-private[sql] case class ExpressionColumnNode private(
+@DeveloperApi
+case class ExpressionColumnNode private(
```

Comment:
You could use `org.apache.spark.sql.classic.ClassicConversions` for this. I am not saying that you should, some people don't like implicits.

Reply:
So it doesn't do both the directions of the conversion and, as you point out, it's also an implicit conversion which (in my opinion) is some pretty unnecessary magic for this.

Reply:
In this case implicits provide a way for a developer to make a minimal amount of changes, that is why they are there. I am generally not a big fan, but in this case they do make migration a lot easier.

You could call `org.apache.spark.sql.classic.ClassicConversions.column` directly if you want to avoid implicits.